### ISSUE TITLE
Trailing commas inside arrays

### ISSUE THREAD

**Issue by jgsojo** [palantir/tslint#1469]
### Bug Report

- __TSLint version__: `3.13.0`
- __TypeScript version__: `1.8.10`
- __Running TSLint via__: `Node.js API`

#### TypeScript code being linted

```ts
param: string[] = [
"a",
"b"
]
```

with `tslint.json` configuration:

```json
"trailing-comma": [true, {"multiline": "always", "singleline": "never"}],
```

#### Actual behavior

Err: Missing trailing comma

#### Expected behavior

Should not mark as error because in ie8(es3) the arrays will differ:
```ts
["a","b",] = ["a", "b", undefined]
["a","b"] = ["a","b"]
```

**Comment by JKillian** [comment]
Hi @jgsojo, you're seeing this error because of a TypeScript bug I believe: https://github.com/Microsoft/TypeScript/issues/7938. TS should presumably omit the trailing comma when you're targeting ES3 and IE8,
